be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -25,156 +25,128 @@ object RollCallHandler extends MessageHandler {
     }
   }
 
   def handleCloseRollCall(rpcRequest: JsonRpcRequest): GraphMessage = {
-    val ask: Future[GraphMessage] = dbAskWritePropagate(rpcRequest)
-    Await.result(ask, duration) match {
-      case Left(_) =>
-        rpcRequest.getParamsMessage match {
-          case Some(message: Message) =>
-            val data: CloseRollCall = message.decodedData.get.asInstanceOf[CloseRollCall]
-
-            // creates a channel for each attendee (of name /root/lao_id/social/PublicKeyAttendee), returns a GraphMessage
-            def createAttendeeChannels(attendees: List[PublicKey]): GraphMessage = {
-              val listAttendeeChannels: List[(Channel, ObjectType.ObjectType)] = data.attendees.map {
-                attendee => (generateSocialChannel(rpcRequest.getParamsChannel, attendee), ObjectType.CHIRP)
-              }
-
-              val askCreateChannels = dbActor ? DbActor.CreateChannelsFromList(listAttendeeChannels)
-
-              Await.ready(askCreateChannels, duration).value match {
-                case Some(Success(_))                        => Left(rpcRequest)
-                case Some(Failure(ex: DbActorNAckException)) => Right(PipelineError(ex.code, s"handleCloseRollCall failed : ${ex.message}", rpcRequest.getId))
-                case reply                                   => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, s"handleCloseRollCall failed : unexpected DbActor reply '$reply'", rpcRequest.getId))
-              }
-            }
-            val laoChannel: Option[Hash] = rpcRequest.getParamsChannel.decodeChannelLaoId
-            laoChannel match {
-              case None => Right(PipelineError(
-                  ErrorCodes.SERVER_ERROR.id,
-                  s"There is an issue with the data of the LAO",
-                  rpcRequest.id
-                ))
-              case Some(_) =>
-                val combined = for {
-                  _ <- dbActor ? DbActor.WriteLaoData(rpcRequest.getParamsChannel, message, None)
-                  _ <- dbActor ? DbActor.WriteRollCallData(laoChannel.get, message)
-                } yield ()
-
-                Await.ready(combined, duration).value match {
-                  case Some(Success(_))                        => createAttendeeChannels(data.attendees)
-                  case Some(Failure(ex: DbActorNAckException)) => Right(PipelineError(ex.code, s"handleCloseRollCall failed : ${ex.message}", rpcRequest.getId))
-                  case reply                                   => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, s"handleCloseRollCall failed : unexpected DbActor reply '$reply'", rpcRequest.getId))
-                }
-            }
-          case _ => Right(PipelineError(
-              ErrorCodes.SERVER_ERROR.id,
-              s"Unable to handle lao message $rpcRequest. Not a Publish/Broadcast message",
-              rpcRequest.id
-            ))
+    val laoChannel: Option[Hash] = rpcRequest.getParamsChannel.decodeChannelLaoId
```

Comment:
I don't know how much you've already thought about it, the function looks already way cleaner but maybe it could be stripped down a bit more still, I'll try to figure out something

Reply:
I added a new function `checkLaoChannel` which does the same as `checkParameters`.